cw_decred/lib/wallet_addresses.dart

```diff
@@ -0,0 +1,46 @@
+import 'dart:convert';
+
+import 'package:cw_core/wallet_addresses.dart';
+import 'package:cw_core/wallet_info.dart';
+import 'package:cw_decred/api/libdcrwallet.dart' as libdcrwallet;
+
+class DecredWalletAddresses extends WalletAddresses {
+  DecredWalletAddresses(WalletInfo walletInfo) : super(walletInfo);
+
+  String currentAddr = '';
+
+  @override
+  String get address {
+    final cAddr = libdcrwallet.currentReceiveAddress(walletInfo.name) ?? '';
+    if (cAddr != '') {
+      currentAddr = cAddr;
+    }
+    return currentAddr;
+  }
+
+  String generateNewAddress() {
+    final nAddr = libdcrwallet.newExternalAddress(walletInfo.name) ?? '';
+    if (nAddr != '') {
+      currentAddr = nAddr;
+    }
+    return nAddr;
+  }
+
+  List<String> addresses() {
+    final res = libdcrwallet.addresses(walletInfo.name);
+    final addrs = (json.decode(res) as List<dynamic>).cast<String>();
+    return addrs;
+  }
+
+  @override
+  set address(String addr) {}
+
+  @override
+  Future<void> init() async {}
+
+  @override
+  Future<void> updateAddressesInBox() async {}
+
+  @override
+  Future<void> saveAddressesInBox() async {}
```

Comment:
Unsure exactly what's needed here but removing the `saveAddressesInBox` override and calling that in `updateAddressesInBox()` for now.

We should stick unused addresses in `addressesMap`?

Reply:
If I use an address from the address book, it inserts a name. Is this a result of mistakes here?
![image](https://github.com/user-attachments/assets/57e6e1eb-f55d-4c19-aa47-49573fb30fc5)

Reply:
yup, on this [line](https://github.com/cake-tech/cake_wallet/pull/1322/files#diff-f20234d9bddc253488cfd9721780cb9446ddbb54fc8b7a421bd349e30e9e93feR287)
and if there is any other line that uses the `out.address` it should be like this
```Dart
out.isParsedAddress ? out.extractedAddress! : out.address
```
like this [here](https://github.com/cake-tech/cake_wallet/blob/main/cw_bitcoin/lib/electrum_wallet.dart#L1022)

Reply:
and for the unimplemented functions, for example, the setter `set address` is going to be used if a user manually selects an address from the address list in the receive screen, you can try it with a bitcoin wallet for example, were you can go to the receive screen, tap on addresses at the bottom, and it will display a list of addresses which are the ones that are used and the ones that are not yet used, and the user can select a different address if he wants

Reply:
Thank you! Will update this area.

Reply:
Is it ok if we don't give out too many unused choices? Because we use an HD wallet, we increase the address index whenever one is used on chain, so if a user were to choose the address at say index 20 when they are at 10, we start to have gaps and "wasted" addresses. We can look at all of these individually, but it takes a bit of back and forth.

The easiest solution is to assume all addresses before the index are used, and all after are unused.